qiskit/algorithms/optimizers/nlopts/nloptimizer.py

```diff
@@ -82,7 +78,7 @@ def get_support_level(self):
 
     @property
     def settings(self):
-        return {"max_evals": self._options.get("max_evals", 1000)}
+        return {"max_evals": self._max_evals}
```

Comment:
This was expected to be in options - I guess we did it for consistency with the way we did things with the scipy optimizer. Line 116 below will now not be doing what is expected.

Reply:
Yeah but isn't it kind of weird to write an easy thing in such a nested fashion? 😅

Reply:
The options was done under locals so as is was common code and you simply listed the arguments that were considered part of options. That wasn't really my point. Later in the call to minimize (which is the line I reference above) it extracts max_evals from options where it no longer exists. Hence if its going to have its own instance var that line needs to get changed too.